### PR TITLE
refactor(redux): add react-query for incidents and patients search

### PR DESCRIPTION
**Pull request by jackcmeyer**
Fixes #1995 

# TODO: 
- [ ] patients
- [ ] scheduling
- [x] incidents
- [ ] imaging
- [ ] medication
- [ ] labs

### CODE REVIEW

**Comment by fox1t** [comment]
This is huge! <3

File: src/App.tsx [hunk 1]
```diff
@@ -33,14 +34,17 @@ const App: React.FC = () => {
...
+    <>
+      <BrowserRouter>
+        <Suspense fallback={<Spinner color="blue" loading size={[10, 25]} type="ScaleLoader" />}>
+          <Switch>
+            <Route exact path="/login" component={Login} />
+            <Route path="/" component={HospitalRun} />
+          </Switch>
+        </Suspense>
+      </BrowserRouter>
+      <ReactQueryDevtools initialIsOpen={false} />
```
Comment (fox1t) [inline src/App.tsx:46]: Is it a way of making this available only in "dev mode"? I am not fammiliar with react-query-devtools at all.

**Comment by fox1t** [comment]
All seems more clear to me when comes to "fetching" data.

File: src/App.tsx (hunk 1 above)
Comment (blestab) [reply src/App.tsx:46]: > 
> 
> Is it a way of making this available only in "dev mode"? I am not fammiliar with react-query-devtools at all.

I thought so too then checked its npm page and it says:

_from the npm page React Query Devtools are not imported and used when process.env.NODE_ENV === 'production', so you don't need to worry about excluding them during a production build._

**Comment by tehKapa** [comment]
LGTM though coveralls alert

**Pull request merged**